### PR TITLE
fix verify-file-perms.sh

### PR DESCRIPTION
**Pull request by BenTheElder**
- work with macOS find (-perm)
- ignore scripts in .git

### CODE REVIEW

**Comment by BenTheElder** [comment]
ugh, the manpage lies. http://man7.org/linux/man-pages/man1/find.1.html
+ is supposed to be deprecated not removed.

**Comment by cblecker** [comment]
/lgtm
/approve

Tested locally on my mac.

File: hack/verify-file-perms.sh [hunk 1]
```diff
@@ -17,7 +17,19 @@ set -o errexit
...
+# we're not actually running a search so SC2185 doesn't apply
+# shellcheck disable=SC2185
+if ! find --version >/dev/null 2>&1; then
+  desired_perm="+111"
+fi
+
+# find all files named *.sh (approximate shell script detection ...)
+# - ignoring .git
+# - that are not executable by all
+files=$(find . -type f -name '*.sh' -not -perm "${desired_perm}" -not -path './.git/*')
```
Comment (BenTheElder) [inline hack/verify-file-perms.sh]: also fixed "*.sh" -> '*.sh', we want the pattern, not a shell glob

File: hack/verify-file-perms.sh [hunk 2]
```diff
@@ -17,7 +17,19 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-files=$(find . -name "*.sh" \! -perm /a+x)
+# BSD sed doesn't have --version, and needs + instead of /
+# GNU sed deprecated and removed +
+desired_perm="/111"
+# we're not actually running a search so SC2185 doesn't apply
+# shellcheck disable=SC2185
```
Comment (fejta) [inline hack/verify-file-perms.sh]: Can't we inline this with the if so it just disables that line rather than everything after it?

Comment (BenTheElder) [reply hack/verify-file-perms.sh]: this doesn't disable the whole file, to disable the whole file you have to put it directly after the shebang https://github.com/koalaman/shellcheck/wiki/Ignore#ignoring-all-instances-in-a-file-044

Comment (BenTheElder) [reply hack/verify-file-perms.sh]: except after a shebang, or as a CLI flag, disables are specific to the following line

**Comment by Katharine** [comment]
/retest

:roll_eyes:

**Pull request merged**